joint_trajectory_controller/src/joint_trajectory_controller_parameters.yaml

```diff
@@ -66,6 +66,13 @@ joint_trajectory_controller:
       one_of<>: [["splines", "none"]],
     }
   }
+  cmd_timeout: {
+    type: double,
+    default_value: 0.0, # seconds
+    description: "Timeout after which the input command is considered stale.
+     Timeout is counted from the end of the trajectory (the last point).
```

Comment:
Hello!
Thanks for the PR.
In the code, you are verifying the command timeout at every sample point right? Why not w.r.t to the total trajectory duration w.r.t to current trajectory duration time

Moreover, why not rename it to `trajectory_timeout`?, because `cmd_timeout` might give an impression about every command point of the sent goal or message.

What do you think?

Reply:
The requested feature here is to stop the robot from doing dangerous things if, e.g., the communication is broken suddenly or the node sending the new trajectories is crashed
> If using velocity as command interface, and the last velocity point is not zero -> the robot would move forever without stopping if no new trajectory is received.

This is very close to the `goal_time` tolerance, where additionally it is checked if the goal itself is reached. 

For both timeouts the same time information 
```c++
      const rclcpp::Time traj_start = (*traj_point_active_ptr_)->get_trajectory_start_time();
      const rclcpp::Time traj_end = traj_start + start_segment_itr->time_from_start;
```
is used, where the iterators are updated with every call of the `sample` method. When the sample method reaches the last point, `traj_end` is the time instance of the _current trajectory duration time_ as you mentioned. This means only, that this time was never calculated explicitly. Or am I missing something?

Acc. to the naming: As the point of this PR is the lack of a new command coming in, I'll stick to `cmd_timeout`. Furthermore, this is similar to the `cmd_vel_timeout` parameter of the _diff_drive_controller_

Reply:
@christophfroehlich Thanks for the clear explanation. Yes, now the `cmd_timeout` makes sense. 

If this is the case, then I have another question. Why not use `allow_nonzero_velocity_at_trajectory_end` parameter from [your other PR](https://github.com/ros-controls/ros2_controllers/pull/567) along with a condition of `cmd_timeout > 0`, just not to trigger it if the `goal_time_tolerance` is set higher than the `cmd_timeout`.

Ideally, the timeout needs to be higher than the goal tolerance. So, maybe we need to either document it better or have a warning printed when `cmd_timeout` is lower than `goal_time_tolerance` or maybe we can consider the command timeout adding goal_time_tolerance.

Reply:
You are right, both timeouts can be competing each other. I updated the docs, added a check in the `on_configure` method and test it.

Reply:
Awesome. Now, it looks good. Thank you. :)

Reply:
I was thinking again if my explanation is correct and extended the timeout test. It failed first, because I missed that I must not check the timeout before the last point of the message was reached. (time difference is counting from the last point reached and not from the to-be-reached point).
Fix + extended tests are pushed.

Reply:
Perfect! Now, it makes sense after using the before_last_point